bugsnag-android-core/src/main/java/com/bugsnag/android/Configuration.java

```diff
@@ -0,0 +1,833 @@
+package com.bugsnag.android;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * User-specified configuration storage object, contains information
+ * specified at the client level, api-key and endpoint configuration.
+ */
+@SuppressWarnings("ConstantConditions") // suppress warning about making redundant null checks
+public class Configuration implements CallbackAware, MetadataAware, UserAware {
+
+    final ConfigImpl impl;
+
+    /**
+     * Constructs a new Configuration object with default values.
+     */
+    public Configuration(@NonNull String apiKey) {
+        impl = new ConfigImpl(apiKey);
+    }
+
+    /**
+     * Loads a Configuration object from values supplied as meta-data elements in your
+     * AndroidManifest.
+     */
+    @NonNull
+    public static Configuration load(@NonNull Context context) {
+        return ConfigImpl.load(context);
+    }
+
+    @NonNull
+    static Configuration load(@NonNull Context context, @NonNull String apiKey) {
+        return ConfigImpl.load(context, apiKey);
+    }
+
+    private void warn(String property) {
```

Comment:
Following the guidance to the letter, this would log an error.

Reply:
Updated to error log level.